cmd/containerboot/main.go

```diff
@@ -295,9 +297,11 @@ authLoop:
 			log.Fatalf("tailscaled left running state (now in state %q), exiting", *n.State)
 		}
 		if n.NetMap != nil {
-			if cfg.ProxyTo != "" && len(n.NetMap.Addresses) > 0 && deephash.Update(&currentIPs, &n.NetMap.Addresses) {
-				if err := installIPTablesRule(ctx, cfg.ProxyTo, n.NetMap.Addresses); err != nil {
-					log.Fatalf("installing proxy rules: %v", err)
+			newCurrentIPs := deephash.Hash(&n.NetMap.Addresses)
+			ipsHaveChanged := newCurrentIPs != currentIPs
```

Comment:
You never set currentIPs

Reply:
They get set here https://github.com/tailscale/tailscale/pull/9031/commits/4fe28d58331cd365ba884e48714d9d6d801820a9#diff-700830ac79a41da42cd3a82065f0fee3003a4390eead4db1de4bf30557a06bd6R324
It's the same logic where they are set after the loop has ran once, but it's moved lower so that we use the same currentIPs for egress proxy